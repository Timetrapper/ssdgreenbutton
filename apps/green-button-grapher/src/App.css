html, body {
  height: 100%;
  width: 100%;
}

body {
  box-sizing: border-box;
  background-color: #c7d6ce;
  font-family: "Trebuchet MS", "Lucida Grande", "Lucida Sans Unicode", "Lucida Sans", Tahoma, sans-serif;
  padding: 0 20px 0 20px;
}

.loading {
  padding: 20px;
  font-size: 20px;
  text-align: center;
}

.address {
  text-align: center;
  padding: 20px;
}

.address > h1 {
  font-size: 20px;
}

.source-code {
  text-align: center;
  padding: 0 0 10px 0;
}

