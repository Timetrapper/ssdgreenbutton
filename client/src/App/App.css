.App {
  text-align: center;
  background: #f5f5f5;;
}

.App-header {
  background-color:black;
  background: linear-gradient(to bottom, rgba(0,0,0,1) 0%, rgba(0,0,0,1) 60%, rgba(0,225,60,1) 80%, rgba(245,245,245,1) 82%, rgba(245,245,245,1) 100%);
  height: 200px;
  width: 100%;
  top: 0;
  color: white;
  align-content: vertical;
  margin-bottom: 10px;
}

.App-green-button-image-header {
  animation-duration: 3s;
  animation-name: bounceIn;
  animation-iteration-count: 1;
  float: left;
  margin: 0px;
  padding-top: 20px;
  padding-left: 40px;
  height: 180px;
  width: auto;
  overflow: hidden;
}

.App-title-image-header {
  float: right;
  margin: 0px;
  padding-top: 20px;
  padding-right: 160px;
  position: relative;
  height: 80px;
  width: auto;
}

.Main-body-container {
  min-height: 100vh;
  overflow: hidden;
  display: block;
  position: relative;
}

.Bar-chart-wrapper {
  padding: 40px;
  margin: 40px;
}

.App-button-wrapper {
  align-content: center;
  float: center;
  padding: 10px;
  margin: 20px;
}

.App-buttons {
  animation-duration: 3s;
  animation-name: bounceIn;
  animation-iteration-count: 1;
  font-size: large;
  float: center;
  padding: 10px;
  margin-left: 5px;
  margin-right: 5px;
  margin-top: 5px;
  margin-bottom: 5px;
}

.App-upload-button-wrapper {
  align-content: center;
  float: center;
  padding: 10px;
  margin: 20px;
}

.App-upload-button {
  font-size: large;
  float: center;
  padding: 10px;
  margin-left: 5px;
  margin-right: 5px;
  margin-top: 5px;
  margin-bottom: 5px;
}

.App-footer {
  background: #000000;
  background: linear-gradient(0deg, rgba(0,0,0) 90%,rgba(0,225,59,1) 100%);
  height: 40px;
  bottom: 0;
  width: 100%;
  padding: 20px;
  width: 100%;
  bottom:0;
  margin-top: 10px;
}

.App-footer-images-container {
  float: right;
  padding: 0px;
  margin: 0px;
}

.App-plasmatic-logo-footer {
  height: 40px;
  float: center;
  margin-left: 30px;
  margin-right: 30px;
  overflow-x: hidden;
}

.App-green-button-logo-footer {
  height: 40px;
  float: center;
  margin-left: 30px;
  margin-right: 30px;
  overflow-x: hidden;
}

.App-github-logo-footer {
  height: 40px;
  float: center;
  margin-left: 30px;
  margin-right: 30px;
  overflow-x: hidden;
}

@keyframes bounceIn {
  0% {
    transform: scale(0.1);
    opacity: 0;
  }
  60% {
    transform: scale(1.2);
    opacity: 1;
  }
  100% {
    transform: scale(1);
  }
}