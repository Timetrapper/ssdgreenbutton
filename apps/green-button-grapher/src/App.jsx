import React, { Component } from 'react';

import './App.css';
import ChartButtons from './ChartButtons';
import GraphContainer from './GraphContainer';
import Slider from './Slider';
import FileOpener from './FileOpener';
import FileApiWarning from './FileApiWarning';
import sampleData from './sample-data';
import GreenButtonJson from './green-button-json';
import INITIAL_APP_STATE from './initial-app-state';

class App extends Component {
  constructor(props) {
    super(props);
    this.state = INITIAL_APP_STATE;
  }

  componentDidMount() {
    this.handleFileLoaded(sampleData);
  }

  get hasFileApi() {
    return !!window.File && !!window.FileReader && !!window.FileList && !!window.Blob;
  }

  handleFileLoaded = (xmlString) => {
    let greenButtonJson = greenButtonJsonFromXmlString(xmlString);
    this.setState({
      loading: false,
      address: greenButtonJson.address,
      intervals: greenButtonJson.intervals
    });
  }

  handleFileSelected = () => {
    this.setState({ loading: true });
  }

  handleSliderMoved = (multiplier) => {
    this.setState({ multiplier: multiplier });
  }

  changeChartType = (chartType) => {
    this.setState({ chartType: chartType });
  }

  changeTimeCut = (timeCut) => {
    this.setState({ timeCut: timeCut });
  }

  render() {
    if (!this.hasFileApi) return <FileApiWarning />;
    if (this.state.loading) return <div className="loading">Loading...</div>;

    return (
      <div className="App">
        <div className="address"><h1>Green Button Data for {this.state.address}</h1></div>
        <GraphContainer
          intervals={this.state.intervals}
          multiplier={this.state.multiplier}
          chartType={this.state.chartType}
          timeCut={this.state.timeCut}
        />
        <Slider
          multiplier={this.state.multiplier}
          handleSliderMoved={this.handleSliderMoved}
        />
        <ChartButtons
          changeChartType={this.changeChartType}
          changeTimeCut={this.changeTimeCut}
          currentChartType={this.state.chartType}
          currentTimeCut={this.state.timeCut}
        />
        <FileOpener
          handleFileSelected={this.handleFileSelected}
          handleFileLoaded={this.handleFileLoaded}
        />
        <div className="source-code">
          <a href="https://github.com/mtmckenna/green-button-grapher">Green Button Grapher source code</a>
        </div>
      </div>
    );
  }
}

function greenButtonJsonFromXmlString(xmlString) {
  let xml = new DOMParser().parseFromString(xmlString, 'text/xml');
  return new GreenButtonJson(xml);
}

export default App;
