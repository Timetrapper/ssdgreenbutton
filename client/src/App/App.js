import './App.css';
import { Bar } from 'nivo';
import { withAlert } from 'react-alert';
import { withStyles, MuiThemeProvider, createMuiTheme } from 'material-ui/styles';
import classNames from 'classnames';
import githubLogo from './github.png';
import greenButtonLogo from './green-button.png';
import plasmaticLogo from './plasmatic.png';
import PropTypes from 'prop-types';
import RaisedButton from 'material-ui/RaisedButton';
import React, { Component } from 'react';

export default class App extends React.Component {
  constructor() {
    super();
    this.state = { data: [
      {
        id: "A",
        value: 10
      },
      {
        id: "B",
        value: 14
      },
      {
        id: "C",
        value: 20
      },
      {
        id: "D",
        value: 18
      },
      {
        id: "E",
        value: 12
      },
      {
        id: "F",
        value: 14
      },
      {
        id: "G",
        value: 20
      },
      {
        id: "H",
        value: 18
      },
      {
        id: "I",
        value: 12
      },
      {
        id: "J",
        value: 14
      },
      {
        id: "K",
        value: 20
      },
      {
        id: "L",
        value: 18
      }
    ] };
  }

  render() {

    return (
      <div className="App">

        <header className="App-header">
          <h1 className="App-title">Green Button Data</h1>
          <img src={greenButtonLogo} className="App-green-button-logo" alt="greenButtonLogo" />
        </header>

        <div>  
          <RaisedButton
              name="file" 
              id="file"
              class="inputfile"
              className="App-buttons"
              containerElement='label'
              label='UPLOAD MY GREEN BUTTON XML FILE HERE'>
                <input for="file" type="file"/>
          </RaisedButton>
        </div> 

        <div class="Bar-chart-wrapper">
          <Bar
            data={this.state.data}
            keys={this.state.data.value} // This isn't working...
            width={600}
            height={400}
            margin={{
              "top": 20,
              "right": 10,
              "bottom": 20,
              "left": 10
            }}
            indexBy='id'
            padding={0.2}
            //axisBottom={{legend: "USERS", legendPosition: "center"}}
            colors={["#61cdbb", "#97e3d5", "#f47560", "#e25c3b"]}
            axisLeft={{
              // using custom function
              format: d => `${d} kWh`
            }}
          />
        </div>

        <RaisedButton className="App-buttons"
              onClick={() => {(window.confirm('Display 24 Hour Period')) } }
        >Display 24 Hours
        </RaisedButton>

        <RaisedButton className="App-buttons"
              onClick={() => {(window.confirm('Display 7 Day Period')) } }
        >Display 7 Days
        </RaisedButton>

        <RaisedButton className="App-buttons"
              onClick={() => {(window.confirm('Display 365 Day Period')) } }
        >Display 30 Days
        </RaisedButton>

        <RaisedButton className="App-buttons"
              onClick={() => {(window.confirm('Display 365 Day Period')) } }
        >Display 365 Days
        </RaisedButton>

        <RaisedButton className="App-buttons"
              onClick={() => {(window.confirm('Display All-Time')) } }
        >Display All-Time
        </RaisedButton>

        <footer className="App-footer">
          <a href="https://www.plasmatic.ai/">
            <img src={plasmaticLogo} className="App-plasmatic-logo" alt="plasmaticLogo"/>
          </a>
          <a href="https://www.github.com/"> {/*add respository*/}
            <img src={githubLogo} className="App-github-logo" alt="githubLogo"/>
          </a>
        </footer>

      </div> 
    );
  }
}