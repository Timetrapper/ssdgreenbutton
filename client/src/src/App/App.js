import React, { Component, Fragment } from 'react';
import { Header, Footer } from '../_layouts';
import logo from '../logo.svg';
import './App.css';

class App extends Component {
  render() {
    return <Fragment> 
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h1 className="App-title">Welcome to React</h1>
          <Header />
        </header>
        <p className="App-intro">
          To get started, edit <code>src/App.js</code> and save to reload.
        </p>
      </div>
    < Footer />
    </Fragment>

  }
}

export default App;
