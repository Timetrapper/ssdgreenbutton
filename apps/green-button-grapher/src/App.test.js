import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import nightmare from 'nightmare'
require('nightmare-upload')(nightmare);
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import url from 'url'
import visit from '../helpers/visit'

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
});

it('renders correctly', () => {
  const tree = renderer.create(
    <App />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

describe('When the app boots', function () {
  let server;
  beforeEach(function() { server = runStaticServer(); });
  afterEach(function() { server.close(); });

  test('it displays test data and can parse a new file', async function () {
    const SECOND_FILE_PATH = '/public/data/Mountain_Single_family_Jan_1_2011_to_Jan_1_2012.xml';
    const FULL_SECOND_FILE_PATH = `${process.cwd()}${SECOND_FILE_PATH}`;

    let page = visit('/index.html');

    let initialText = await page
      .wait(() => document.body.textContent.includes('123 SAMPLE ST BERKELEY CA 94707-2701'))
      .click("button[data-type='chart-type-cost']")
      .evaluate(() => document.body.textContent);

    expect(initialText).toContain('Total: $1.30');
    expect(initialText).toContain('Peak: $0.45');

    page.upload("input[type='file']", FULL_SECOND_FILE_PATH);

    let textAfterUpload = await page
      .wait(() => document.body.textContent.includes('Mountain Single-family'))
      .click("button[data-type='chart-type-power-usage']")
      .evaluate(() => document.body.textContent)
      .end();

    expect(textAfterUpload).toContain('Total: 24,380 kWh');
    expect(textAfterUpload).toContain('Peak: 5,339 kWh');
  });
});

describe('When loading a file', function() {
  test('it shows the loading screen', function() {
    const wrapper = shallow(<App />);
    wrapper.setState({ loading: false });
    expect(wrapper.text()).not.toContain('Loading');
    wrapper.setState({ loading: true });
    expect(wrapper.text()).toContain('Loading');
  });
});

function runStaticServer() {
  let express = require('express');
  let app = express();
  app.use(express.static('build'));
  return app.listen(3001);
}
