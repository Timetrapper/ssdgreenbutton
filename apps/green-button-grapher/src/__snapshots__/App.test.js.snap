// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="App"
>
  <div
    className="address"
  >
    <h1>
      Green Button Data for 
      123 SAMPLE ST BERKELEY CA 94707-2701
    </h1>
  </div>
  <div>
    <div
      className="graph-wrapper"
    >
      <canvas />
    </div>
    <div
      className="results"
    >
      <div>
        <div>
          <strong>
            Total:
          </strong>
           
          9,995 kWh
        </div>
        <div>
          <strong>
            Peak:
          </strong>
           
          3,469 kWh
        </div>
      </div>
    </div>
  </div>
  <div>
    <input
      className="slider"
      max="2"
      min="0"
      onChange={[Function]}
      step="0.01"
      type="range"
      value={1}
    />
  </div>
  <div>
    <div
      className="button-set"
    >
      <fieldset>
        <legend>
          Raw Data
        </legend>
        <button
          className="button "
          data-type="most-recent-24-hours"
          onClick={[Function]}
        >
          Most Recent 24 Hours
        </button>
        <button
          className="button "
          data-type="last-7-days"
          onClick={[Function]}
        >
          Last 7 Days
        </button>
        <button
          className="button "
          data-type="last-30-days"
          onClick={[Function]}
        >
          Last 30 Days
        </button>
        <button
          className="button "
          data-type="all-time"
          onClick={[Function]}
        >
          All Time
        </button>
      </fieldset>
    </div>
    <div
      className="button-set"
    >
      <fieldset>
        <legend>
          Averages
        </legend>
        <button
          className="button button--active"
          data-type="avg-day"
          onClick={[Function]}
        >
          Day
        </button>
        <button
          className="button "
          data-type="avg-weekend-day"
          onClick={[Function]}
        >
          Weekend Day
        </button>
        <button
          className="button "
          data-type="avg-week-day"
          onClick={[Function]}
        >
          Week Day
        </button>
        <button
          className="button "
          data-type="avg-peak-time"
          onClick={[Function]}
        >
          Peak Time
        </button>
      </fieldset>
    </div>
    <div
      className="button-set"
    >
      <fieldset>
        <legend>
          Chart Type
        </legend>
        <button
          className="button "
          data-type="chart-type-cost"
          onClick={[Function]}
        >
          Cost
        </button>
        <button
          className="button button--active"
          data-type="chart-type-power-usage"
          onClick={[Function]}
        >
          Power Usage
        </button>
      </fieldset>
    </div>
  </div>
  <div
    className="file-opener"
  >
    <input
      onChange={[Function]}
      type="file"
    />
  </div>
  <div
    className="source-code"
  >
    <a
      href="https://github.com/mtmckenna/green-button-grapher"
    >
      Green Button Grapher source code
    </a>
  </div>
</div>
`;
